sagemaker/launch/launch_train.py

```diff
@@ -75,24 +80,12 @@ def run_job(input_args, image, unknowargs):
         params["custom-script"] = custom_script
     if model_checkpoint_to_load is not None:
         params["model-checkpoint-to-load"] = model_checkpoint_to_load
-    # We must handle cases like
-    # --target-etype query,clicks,asin query,search,asin
-    # --feat-name ntype0:feat0 ntype1:feat1
-    unknow_idx = 0
-    while unknow_idx < len(unknowargs):
-        assert unknowargs[unknow_idx].startswith("--")
-        sub_params = []
-        for i in range(unknow_idx+1, len(unknowargs)+1):
-            # end of loop or stand with --
-            if i == len(unknowargs) or \
-                unknowargs[i].startswith("--"):
-                break
-            sub_params.append(unknowargs[i])
-        params[unknowargs[unknow_idx][2:]] = ' '.join(sub_params)
-        unknow_idx = i
-
-    print(f"Parameters {params}")
-    print(f"GraphStorm Parameters {unknowargs}")
+    unknown_args_dict = parse_unknown_gs_args(unknowargs)
+    if len(unknown_args_dict) > 0:
+        params.update(unknown_args_dict)
```

Comment:
If unknown_args_dict is always a dict, we can directly do `params.update(unknown_args_dict)`.
`params.update({})` will not change params.